wire-library/wire-tests/src/commonTest/proto-kotlin/com/squareup/wire/protos3/kotlin/person/Person.kt

```diff
@@ -156,18 +157,20 @@ class Person(
       Person::class, 
       "type.googleapis.com/squareup.protos3.kotlin.person.Person"
     ) {
-      override fun encodedSize(value: Person): Int = 
+      override fun encodedSize(value: Person): Int {
+        var size = value.unknownFields.size
         if (value.name == "") 0
-        else ProtoAdapter.STRING.encodedSizeWithTag(1, value.name) +
+        else size += ProtoAdapter.STRING.encodedSizeWithTag(1, value.name)
         if (value.id == 0) 0
-        else ProtoAdapter.INT32.encodedSizeWithTag(2, value.id) +
+        else size += ProtoAdapter.INT32.encodedSizeWithTag(2, value.id)
         if (value.email == "") 0
-        else ProtoAdapter.STRING.encodedSizeWithTag(3, value.email) +
-        PhoneNumber.ADAPTER.asRepeated().encodedSizeWithTag(4, value.phones) +
-        ProtoAdapter.STRING.asRepeated().encodedSizeWithTag(5, value.aliases) +
-        ProtoAdapter.INT32.encodedSizeWithTag(6, value.foo) +
-        ProtoAdapter.STRING.encodedSizeWithTag(7, value.bar) +
-        value.unknownFields.size
+        else size += ProtoAdapter.STRING.encodedSizeWithTag(3, value.email)
+        size += PhoneNumber.ADAPTER.asRepeated().encodedSizeWithTag(4, value.phones)
+        size += ProtoAdapter.STRING.asRepeated().encodedSizeWithTag(5, value.aliases)
+        size += ProtoAdapter.INT32.encodedSizeWithTag(6, value.foo)
+        size += ProtoAdapter.STRING.encodedSizeWithTag(7, value.bar)
+        return size
+      }
```

Comment:
We need to address that.
Maybe we could generate something like
```kotlin
if (value.name != "") size += ProtoAdapter.STRING.encodedSizeWithTag(1, value.name)
```
instead of
```kotlin
if (value.name == "") 0
else size += ProtoAdapter.STRING.encodedSizeWithTag(1, value.name)
```

Reply:
Should be in `KotlinGenerator` around some `encodedSizeFun` or something.

Reply:
Yes please. The current syntax is not great to look at.